cellfinder_napari/detect_utils.py

```diff
@@ -0,0 +1,186 @@
+from dataclasses import dataclass, asdict
+from pathlib import Path
+
+import napari
+from cellfinder_core.main import main as cellfinder_run
+from imlib.cells.cells import Cell
+from napari.qt.threading import thread_worker
+
+from cellfinder_napari.utils import cells_to_array
+
+
+@dataclass
+class InputContainer:
+    """Base data class to group inputs into a container, with persistent defaults"""
+
+    persistent_defaults = (
+        {}
+    )  # some elements need to be remembered without having to create a new instance, for reset button
```

Comment:
I'm assuming this is a `dict` and not a `tuple[dict]`?
```suggestion
    # some elements need to be remembered without having to create a new instance, for reset button
    persistent_defaults = {}
```

Reply:
Yup - good catch. I've refactored and simplified this, so this now looks completely different..